app/lib/history/models.dart

```diff
@@ -197,22 +229,19 @@ class UploaderChanged implements HistoryEvent {
     return '$actor has changed uploaders: ${changes.join(' and ')}.';
   }
 
-  @override
   Map<String, dynamic> toJson() => _$UploaderChangedToJson(this);
 }
 
 @JsonSerializable()
 class AnalysisCompleted implements HistoryEvent {
-  static const _type = 'analysis-completed';
-
   final bool hasErrors;
 
   final bool hasPlatforms;
 
   AnalysisCompleted({this.hasErrors, this.hasPlatforms});
 
-  @override
-  String getType() => _type;
+  factory AnalysisCompleted.fromJson(Map<String, dynamic> json) =>
+      _$AnalysisCompletedFromJson(json);
```

Comment:
What do you need this for?

Reply:
Can't @JsonSerializable magically handle subtypes?

Reply:
this is not a subtype, this is a composition, and it needs to have fromJson and toJson to work with it